graph/pull.go

```diff
@@ -60,7 +60,8 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		return err
 	}
 
-	r, err := registry.NewSession(imagePullConfig.AuthConfig, registry.HTTPRequestFactory(imagePullConfig.MetaHeaders), endpoint, true)
+	client := registry.HTTPClient(&registry.DockerHeaders{registry.DefaultTransport(registry.ReceiveTimeout, endpoint.IsSecure), imagePullConfig.MetaHeaders})
```

Comment:
Might want to break this up over a few lines to let it breath (readability) and bring up your SLOC stats. ;)